app/main/posts/views/post-filters.service.js

```diff
@@ -18,19 +18,73 @@ function PostFiltersService(_, FormEndpoint, TagEndpoint, $q) {
         getQueryParams: getQueryParams,
         getFilters: getFilters,
         setFilters: setFilters,
+        setFilter: setFilter,
         clearFilters: clearFilters,
         clearFilter: clearFilter,
+        clearFilterFromArray: clearFilterFromArray,
         hasFilters: hasFilters,
         getActiveFilters: getActiveFilters,
-        getCleanActiveFilters: getCleanActiveFilters,
+        getUIActiveFilters: getUIActiveFilters,
         setMode: setMode,
         getMode: getMode,
         getModeId: getModeId,
+        getModeEntity: getModeEntity,
         countFilters: countFilters,
+        cleanRemovedValuesFromObject: cleanRemovedValuesFromObject,
+        addIfCurrentObjectMatchesOriginal: addIfCurrentObjectMatchesOriginal,
         reactiveFilters: true,
         qEnabled: false
     };
 
+    /**
+     *
+     *  original = {status: ['draft', 'review', 'archive']};
+     *  current = {status: ['draft', 'review']}
+     *  uiFiltersCurrent = {status: ['draft', 'review', 'archive']}
+     *  Original has 3 filters, then 1 was removed, so current has 2. uiFiltersCurrent has 3 again because the user re-added it.
+     *  We need to compare  original and uiFIltersCurrent.
+     *  If `uiFiltersCurrent` has values that are present in  `original`,
+     *  we need to add them to `current` (*some other function will remove them from uiFiltersCurrent too)
+     *
+     * @param current
+     * @param original
+     */
+    function addIfCurrentObjectMatchesOriginal(currentSS, originalSS, uiObj) {
+        //find filters in current that are part of the original object
+        var ret =  _.mapObject(originalSS, function (obj, key) {
+            if (!_.isArray(obj)) {
+                if (originalSS[key] === uiObj[key]) {
+                    currentSS[key] = uiObj[key];
+                    return currentSS[key];
+                }
+            }
+            // get the array values correctly
+            return _.without(_.flatten(_.zip(currentSS[key], _.filter(uiObj[key], function (val) {
+                return currentSS[key].indexOf(val) < 0;
+            }))), undefined);
+        });
+        return ret;
+    }
+
+    /**
+     * Looks for keys that are NOT present in currentFilters but that are in the savedSearch filters (which means they are removed)
+     * and removes them from the saved search filters array.
+     * Does not handle removal where the key exists but the values are an array and some are missing. Need to fix that.
+     **/
+    function cleanRemovedValuesFromObject(currentFilters, oldFilters) {
```

Comment:
I think this name is slightly better than the previous name so it stays for the moment